lib/config.js

```diff
@@ -1423,6 +1423,21 @@ util.stripYamlComments = function(fileStr) {
 }
 
 /**
+ * Strip all Javascript type comments from JSON strings.
+ *
+ * This method will call util.stripComments but has a simplified regex that
+ * will work with JSON strings.
+ *
+ * @protected
+ * @method stripJSONComments
+ * @param fileString {string} The string to strip comments from
+ * @return {string} The string with comments stripped.
+ */
+util.stripJSONComments = function(fileStr) {
+  return util.stripComments(fileStr, /("[^"]*")/g);
```

Comment:
Sure, just to clarify, this is effectively equivalent to the regex used by `util.stripComments` but is targeted for valid JSON. The one in `util.stripComments` can begin with either `'` or `"` and will capture everything to a closing `'` or `"`. Since this method is only concerned with JSON we don't need to worry about `'` and can focus only on strings that begin and end with `"`. The regex is used  to create valid JSON and remove comments, so skipping over strings within comments is completely acceptable. The test was updated to confirm that valid JSON is generated and I've successfully tested locally with a few different combinations of strings.